spec/lib/submission_builder/ty2024/states/nj/documents/nj1040_spec.rb

```diff
@@ -0,0 +1,53 @@
+require 'rails_helper'
+
+describe SubmissionBuilder::Ty2024::States::Nj::Documents::Nj1040, required_schema: "nj" do
+  describe ".document" do
+    let(:intake) { create(:state_file_nj_intake, filing_status: "single", municipality_code: "0101") }
+    let(:submission) { create(:efile_submission, data_source: intake) }
+    let(:build_response) { described_class.build(submission, validate: false) }
+    let(:xml) { Nokogiri::XML::Document.parse(build_response.document.to_xml) }
+
+    context "single filer" do
+      it "correctly fills answers" do
+        expect(xml.document.at('FilingStatus Single').text).to eq "X"
+      end
+    end
+
+    context "married filing jointly filers" do
+      let(:intake) { create(:state_file_nj_intake, filing_status: "married_filing_jointly") }
+      
+      it "correctly fills spouse-specific answers" do
+        expect(xml.document.at('FilingStatus MarriedCuPartFilingJoint').text).to eq "X"
+      end
+    end
+
+    context "married filing separately filers" do
+      let(:intake) { create(:state_file_nj_intake, :mfs_complete_spouse) }
+      
+      it "correctly fills spouse-specific answers" do
+        expect(xml.document.at('FilingStatus MarriedCuPartFilingSeparate SpouseSSN').text).to eq(intake.spouse.ssn)
+        expect(xml.document.at('FilingStatus MarriedCuPartFilingSeparate SpouseName FirstName').text).to eq(intake.spouse.first_name)
+        expect(xml.document.at('FilingStatus MarriedCuPartFilingSeparate SpouseName MiddleInitial').text).to eq(intake.spouse.middle_initial)
+        expect(xml.document.at('FilingStatus MarriedCuPartFilingSeparate SpouseName LastName').text).to eq(intake.spouse.last_name)
+      end
+    end
+
+    context "qualifying widow/er filers" do
+      let(:intake) { create(:state_file_nj_intake, filing_status: "qualifying_widow") }
+      
+      before do
+        intake.direct_file_data.spouse_date_of_death = "2024-07-30"
+      end
+      
+      it "correctly fills qualifying-widow-specific answers" do
+        expect(xml.document.at('FilingStatus QualWidOrWider QualWidOrWiderSurvCuPartner').text).to eq "X"
+        expect(xml.document.at('FilingStatus QualWidOrWider LastYear').text).to eq "X"
```

Comment:
[pebble] 
- Can we test exhaustively that "TwoYearPrior" is not checked?
- Can we also include a test for when the date is 2023, that TwoYearPrior is checked and LastYear is not
- Can we have a test to handle what happens if the date of death is greater than 2 years prior?

Reply:
I started to test the greater than 2 years prior case, but realized that begs the question of what we should do with filing status in that case. Since the fed requirements for spouse death within last two years are the same, and we should match fed filing status in call cases, is this still a case worth testing for? If so, should we add eligibility offboarding after the data import for this and other potential data quality issues?

Reply:
Marking resolved per team sync discussion